babel/localedata.py

```diff
@@ -41,6 +41,8 @@ def normalize_locale(name):
     Returns the normalized locale ID string or `None` if the ID is not
     recognized.
     """
+    if not name or not isinstance(name, string_types):
```

Comment:
It would probably be better and/or more Pythonic to just cast to the text type here, instead of strictly requiring a string type.